### DISCUSSION TITLE
Add information_line handling to edgelist read/write

### DISCUSSION THREAD

**Discussion by CharJon** [networkx/networkx#4596] [Ideas]
Some well known benchmark libraries in academia offer files in the already implemented edgelist format, but the first line stores meta information about the graph. It consists of "number_of_nodes number_of_edges" (see e.g. [here](http://biqmac.uni-klu.ac.at/biqmaclib.html) or [here](https://www.cc.gatech.edu/dimacs10/archive/streets.shtml)).
I propose adding a flag '''information_line=False''' to the existing functions, that handles this special case.

As far as I know the mete-information line does not have a specific name. I'm open for any other name than "information_line" as I'm calling it right now.

**Comment by rossbar** [comment]
Another possible option would be to add the parameter `skiprows` to mirror the behavior in other popular text readers like `numpy.loadtxt`. This might be a bit more flexible in that the user can skip an arbitrary number of lines, which might be useful for edgelist files that have comments at the top or some other custom header.

IMO this would be more convenient/flexible on the reading end. For writing, it's slightly more work than a keyword arg, but again the main benefit is flexibility. A full round-trip might look something like:

```python
>>> # Writing to an edgelist format that has a summary in the top line
>>> G = nx.path_graph(4)
>>> with open("foo.edgelist", "wb") as ff:
...     ff.write(f"{G.number_of_nodes()} {G.number_of_edges()}\n".encode("utf-8"))
...     nx.write_edgelist(G, ff)
>>> # Reading from such a format
>>> G = nx.read_edgelist("foo.edgelist", skiprows=1)
```

**Reply by CharJon** [reply]
I feel like this comes down to whether or not one sees the mentioned information line as a comment, or more as a feature of the file format. I would compare the information line to the header line of csv files, which functions like pandas.read_csv explicitly take care of. pandas.read_csv additionally offers sciprows. Obviously I am biased on how niche the mentioned libraries are.
